cmd/frontend/internal/app/ui/preview.go

```diff
@@ -0,0 +1,73 @@
+package ui
+
+import (
+	"fmt"
+	"path"
+	"strconv"
+
+	"github.com/sourcegraph/sourcegraph/internal/lazyregexp"
+)
+
+var (
+	singleLineRegexp     = lazyregexp.New(`L(\d+)`)
+	multiLineRangeRegexp = lazyregexp.New(`L(\d+)-(\d+)`)
```

Comment:
Make sure to also handle ranges with columns, as are generated when clicking a symbol ([example](https://sourcegraph.com/github.com/sourcegraph/codeintellify/-/blob/src/hoverifier.ts#L165:13)). It doesn't have to influence the preview (it could in the future to add symbol icons), but the regex should still be able to extract the line if a column is present. This should work:
```suggestion
	singleLineRegexp     = lazyregexp.New(`L(\d+)(?::\d+)?`)
	multiLineRangeRegexp = lazyregexp.New(`L(\d+)(?::\d+)?-(\d+)(?::\d+)?`)
```

Reply:
Also, should these maybe be surrounded with `^` and `$` to not match arbitrary strings containing the pattern?

Reply:
Yes, to both comments 👍